pkg/apis/app/v1/kieapp_types.go

```diff
@@ -486,18 +482,6 @@ type KieAppStatus struct {
 	Deployments olm.DeploymentStatus `json:"deployments"`
 }
 
-type PlatformService interface {
```

Comment:
what was the reason for removing this interface?

Reply:
Because `client.DeleteOptionFunc` is explicitly deleted from `sigs.k8s.io/controller-runtime/pkg/client` v0.2.0+. And this version upgrade is r[equired by the operator-sdk v0.11.0](https://github.com/operator-framework/operator-sdk/releases/tag/v0.11.0). It is a breaking change, which means we have to make breaking changes to the `PlatformService` APIs as well, otherwise the code won't compile. 

I modified v2 of the  `PlatformService` APIs. But I don't see any code is still using v1 of  `PlatformService` APIs.  And I don't think they are part of the CRDs that we expose. So I deleted them. Please let me know if we still want to maintain v1 of the `PlatformService` APIs. Thanks.